clang/test/Sema/warn-extra-cast-function-type-strict.c

```diff
@@ -0,0 +1,42 @@
+// RUN: %clang_cc1 %s -fsyntax-only -Wextra -verify
```

Comment:
If this diagnostic is not already well tested, adding all of these additional unit tests makes sense.

But it looks like there's quite a number of existing tests in other files.  I don't think we should add a new file just to check this diagnostics behavior via `-Wextra` via a whole new file.

Reply:
No I was thinking of adding extra runs for both -Wcast-function-type and -Wextra or should the one for -Wcast-function-type be done in a different issue?

Reply:
I would just remove the newly created clang/test/Sema/warn-extra-cast-function-type-strict.c.  If you have cases that expose bugs in the implementation, that is distinctly a separate issue from adding this warning under -Wextra.

Reply:
Ok. sounds good I will do that.

Reply:
Done. File removed, and a FIXME comment added for the existing test.